sw/vendor/riscv_compliance/riscv-target/opentitan/device/rv32imc/Makefile.include

```diff
@@ -47,28 +49,52 @@ RISCV_NM       ?= ${OT_TOOLS}/riscv32-unknown-elf-nm
 RISCV_READELF  ?= ${OT_TOOLS}/riscv32-unknown-elf-readelf
 RISCV_GCC_OPTS ?= -static -mcmodel=medany -fvisibility=hidden -nostdlib -nostartfiles -g
 
+BUILD_ROOT = $(work_dir_isa)/meson-build
 
 # The compile target recipe re-uses the boot rom library.
 # This will be changed in the future when the compliance tests directly build
 # their own libraries
 # After the libraries are built, the necessary collateral (vmem for verilator, bin
 # for fpga) are created
 COMPILE_TARGET=\
-        make -C $$(OT_SW) SW_DIR=boot_rom $(MAKEARG) RV_TOOLS=$(OT_TOOLS) clean all; \
+  export BUILD_ROOT=$(BUILD_ROOT); \
+	mkdir -p $(BUILD_ROOT); \
+	cd $(OT_ROOT); \
+	./meson_init.sh; \
+	cd -; \
+	ninja -C "$(BUILD_ROOT)/build-out/sw/$(NINJA_TARGET)" all; \
```

Comment:
so i guess this will eventually build all the libraries and difs? Does that include existing tests also?

Reply:
It builds everything. Really not a big deal IMO, but it can be narrowed to just libuart_ot

Reply:
no it's fine, i just wanted to make sure i read it right.